src/Resolution.ts

```diff
@@ -49,160 +43,119 @@ import DnsUtils from './DnsUtils';
  * ```
  */
 export default class Resolution {
-  /** @internal */
-  readonly blockchain: boolean;
-  /** @internal */
-  readonly ens?: Ens;
-  /** @internal */
-  readonly zns?: Zns;
-  /** @internal */
-  readonly cns?: Cns;
-  /** @internal */
-  readonly api?: UdApi;
-
+  private readonly services: NamingService[];
+  private readonly serviceMap: Record<NamingServiceName, NamingService>;
+  
   constructor({
-    blockchain = true,
-    api = DefaultAPI,
-  }: { blockchain?: Blockchain | boolean; api?: API } = {}) {
-    this.blockchain = !!blockchain;
-    if (blockchain) {
-      if (blockchain === true) {
-        blockchain = {};
-      }
-
-      const web3provider = blockchain.web3Provider;
-      if (web3provider) {
-        console.warn(
-          'Usage of `web3Provider` option is deprecated. Use `provider` option instead for each individual blockchain',
-        );
-      }
-
-      const ens = this.normalizeSource(blockchain.ens, web3provider);
-      const zns = this.normalizeSource(blockchain.zns);
-      const cns = this.normalizeSource(blockchain.cns, web3provider);
-
-      if (ens) {
-        this.ens = new Ens(ens);
-      }
-
-      if (zns) {
-        this.zns = new Zns(zns);
-      }
-
-      if (cns) {
-        this.cns = new Cns(cns);
-      }
+    sourceConfig = undefined,
+  }: { sourceConfig?: SourceConfig } = {}) {
 
-    } else {
-      this.api = new UdApi(api);
-    }
+    const cns = (isApi(sourceConfig?.cns) ? new UdApi() : new Cns(sourceConfig?.cns));
```

Comment:
@JohnnyJumper If you need isApi in Resolution only than probably you don't to move this method to utils

Reply:
this method is not going to be used anywhere except resolution. I just need to know if I need to init the class or not. it also makes sure that in else part of the code sourceConfig.cns is not API type.